core/src/main/java/hudson/lifecycle/UnixLifecycle.java

```diff
@@ -50,12 +52,12 @@
  * @since 1.304
  */
 public class UnixLifecycle extends Lifecycle {
-    private JavaVMArguments args;
+    private List<String> args;
     private Throwable failedToObtainArgs;
 
     public UnixLifecycle() throws IOException {
         try {
-            args = JavaVMArguments.current();
+            args = new ArrayList<>(JavaVMArguments.current());
```

Comment:
`JavaVMArguments` delegates to `ProcessTree`, which can sometimes return `Collections.unmodifiableList`. Since we modify the result by removing `--daemon` a few lines down, do a defensive copy.